File: android/app/src/main/java/com/kidscallhome/app/plugins/WidgetDataPlugin.java
// android/app/src/main/java/com/kidscallhome/app/plugins/WidgetDataPlugin.java
// Capacitor plugin for syncing widget data to Android SharedPreferences
// Purpose: Bridge between React app and native Android widget data storage

package com.kidscallhome.app.plugins;

import android.content.Context;
import android.content.Intent;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.kidscallhome.app.widgets.KidsCallHomeWidgetProvider;
import org.json.JSONException;
import org.json.JSONObject;

@CapacitorPlugin(name = "WidgetData")
public class WidgetDataPlugin extends Plugin {

    @PluginMethod
    public void syncWidgetData(PluginCall call) {
        try {
            JSObject data = call.getData();
            
            // Extract widget data from JSObject
            String childId = data.getString("childId");
            String childName = data.getString("childName", "Call Home");
            String childAvatarColor = data.getString("childAvatarColor", "#3B82F6");
            int unreadCount = data.getInteger("unreadCount", 0);
            String lastCallTime = data.getString("lastCallTime");

            // Create WidgetData object
            KidsCallHomeWidgetProvider.WidgetData widgetData = 
                new KidsCallHomeWidgetProvider.WidgetData(
                    childId,
                    childName,
                    childAvatarColor,
                    unreadCount,
                    lastCallTime
                );

            // Store in SharedPreferences via widget provider
            KidsCallHomeWidgetProvider.storeWidgetData(getContext(), widgetData);

            // Trigger widget update
            Intent updateIntent = new Intent(KidsCallHomeWidgetProvider.ACTION_UPDATE_WIDGET);
            getContext().sendBroadcast(updateIntent);

            JSObject result = new JSObject();
            result.put("success", true);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to sync widget data: " + e.getMessage(), e);
        }
    }

    @PluginMethod
    public void updateWidget(PluginCall call) {
        try {
            // Trigger widget update
            Intent updateIntent = new Intent(KidsCallHomeWidgetProvider.ACTION_UPDATE_WIDGET);
            getContext().sendBroadcast(updateIntent);

            JSObject result = new JSObject();
            result.put("success", true);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Failed to update widget: " + e.getMessage(), e);
        }
    }
}

File: android/app/src/main/java/com/kidscallhome/app/plugins/WidgetDataPlugin.kt
// android/app/src/main/java/com/kidscallhome/app/plugins/WidgetDataPlugin.kt
// Capacitor plugin for syncing widget data to Android SharedPreferences
// Purpose: Bridge between React app and native Android widget data storage

package com.kidscallhome.app.plugins

import android.content.Intent
import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin
import com.kidscallhome.app.widgets.KidsCallHomeWidgetProvider

@CapacitorPlugin(name = "WidgetData")
class WidgetDataPlugin : Plugin() {

    @PluginMethod
    fun syncWidgetData(call: PluginCall) {
        try {
            val data = call.data

            // Extract widget data from JSObject
            val childId = data.getString("childId")
            val childName = data.getString("childName", "Call Home") ?: "Call Home"
            val childAvatarColor = data.getString("childAvatarColor", "#3B82F6") ?: "#3B82F6"
            val unreadCount = data.getInt("unreadCount", 0)
            val lastCallTime = data.getString("lastCallTime")

            // Create WidgetData object
            val widgetData = KidsCallHomeWidgetProvider.Companion.WidgetData(
                childId = childId,
                childName = childName,
                childAvatarColor = childAvatarColor,
                unreadCount = unreadCount,
                lastCallTime = lastCallTime
            )

            // Store in SharedPreferences via widget provider
            KidsCallHomeWidgetProvider.storeWidgetData(context, widgetData)

            // Trigger widget update
            val updateIntent = Intent(KidsCallHomeWidgetProvider.ACTION_UPDATE_WIDGET)
            context.sendBroadcast(updateIntent)

            val result = JSObject()
            result.put("success", true)
            call.resolve(result)
        } catch (e: Exception) {
            call.reject("Failed to sync widget data: ${e.message}", e)
        }
    }

    @PluginMethod
    fun updateWidget(call: PluginCall) {
        try {
            // Trigger widget update
            val updateIntent = Intent(KidsCallHomeWidgetProvider.ACTION_UPDATE_WIDGET)
            context.sendBroadcast(updateIntent)

            val result = JSObject()
            result.put("success", true)
            call.resolve(result)
        } catch (e: Exception) {
            call.reject("Failed to update widget: ${e.message}", e)
        }
    }
}

